aip/general/0193.md

```diff
@@ -62,11 +62,120 @@ a [`PreconditionFailure`][PreconditionFailure].
 **Note:** `ErrorInfo` represents a special case. There **must** be exactly one
 `ErrorInfo`. It is required.
 
-### Localization
+### Error messages
 
-Error messages **must** be in English. If a localized error message is also
-required, the service **should** use [`google.rpc.LocalizedMessage`][details]
-in the `details` field.
+For each error, the service **must** populate the `message` field on
+[`google.rpc.Status`][Status]. This error message,
+
+- is a developer-facing, human-readable "debug message"
+- is presented in the service's native language
+- both explains the error and offers an actionable resolution to it
+  ([citation](https://cloud.google.com/apis/design/errors#error_model))
+
+**Note:** Sometimes a service will elect to always present
+`Status.message` in English rather than the application's native
+language so that messages are easily searchable in common knowledge
+bases, such as StackOverflow&trade;.
+
+When introducing a new error, a failure scenario that did not previously
+occur for the service, the payload **must** include `ErrorInfo` and any
+variables found in dynamic segments of the error message **must** be
+present in `ErrorInfo.metadata`. See, "[Dynamic
+variables](#dynamic-variables)".
+
+#### Changing error messages
+
+If your service's API design governing body decides that altering
+`Status.message` represents a backwards incompatible change, then
+`Status.message` **must not** change.
+
+<a name="status-message-warning"></a>
+**Warning:** **Use extreme caution** when changing `Status.message`.
+API consumers often treat this error message as **part of the API
```

Comment:
nit: some of these feels like "Rationale" more than guidance. Guidance at most briefly elaborates on a particular choice, while rationale explains the choice in-depth.

Reply:
Ack, IMO it is important to include this warning here.